### PR TITLE
RATEST-202: Rename refapp 2.x clinical visit and vitals-and-triaging workflow tests

### PR DESCRIPTION
**Pull request by kdaud**
Ticket [ID](https://issues.openmrs.org/browse/RATEST-202)

Description -> Refactor refapp 2.x clinical visit and vitals-and triaging workflow tests

### CODE REVIEW

**Comment by kdaud** [comment]
Waiting for recommendation cc: @sherrif10

**Comment by kdaud** [comment]
Waiting for recommendation cc: @sherrif10

File: .github/workflows/refapp-2x-vitals-and-triaging.yml [hunk 1]
```diff
@@ -47,5 +47,5 @@ jobs:
           echo "cucumber.publish.enabled=true" > src/test/resources/cucumber.properties
           mvn clean install -DskipTests=true
           cp -f src/test/resources/org/openmrs/uitestframework/test-local.properties src/test/resources/org/openmrs/uitestframework/test.properties
-          npm run refapp2VitalsAndTriagingTest
```
Comment (sherrif10) [inline .github/workflows/refapp-2x-vitals-and-triaging.yml:50]: i would leave it the way it was as refapp2vitalsAndTriagingTest. it sounds more better.

File: .github/workflows/refapp-2x-vitals-and-triaging.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: RefApp 2.x Vitals And Triaging Test
```
Comment (sherrif10) [inline .github/workflows/refapp-2x-vitals-and-triaging.yml:1]: Having a name test looks cool and still this is fine as well

**Comment by sherrif10** [comment]
Have fixed changes related with case sensitive in this ticket here https://github.com/openmrs/openmrs-contrib-qaframework/pull/91

**Comment by kdaud** [comment]
> Have fixed changes related with case sensitive in this ticket here #91


It was in my minds to create [RATEST-202 ](https://issues.openmrs.org/browse/RATEST-205) and [RATEST-205](https://issues.openmrs.org/browse/RATEST-205) in order not to mix changes related to  2.x and 3.x refapp so that we can easily track these changes in the future. I would not recommend to mix combine changes of 2.x and 3.x in one commit, and this is not a good move as I reflect in the future perspective of the project.

**Comment by sherrif10** [comment]
Its actually two different tickets if am not mistaken , it seems this is removing tests and mine is handling case sensitive format.

**Comment by kdaud** [comment]
> it seems this is removing tests and mine is handling case sensitive format

You might need to look at the ticket description and get to know what [RATEST-202](https://issues.openmrs.org/browse/RATEST-202) is accomplishing.

**Comment by sherrif10** [comment]
> You might need to look at the ticket description and get to know what [RATEST-202](https://issues.openmrs.org/browse/RATEST-202) is accomplishing.

That is ticket duplication.

**Comment by kdaud** [comment]
IMO I am looking at having our QA-Dashboard having consistency with workflow names that are being displayed. When you look at platform core tests, you see names like Platform Simple Installation, Platform Advanced Installation, etc , as well when you look at 3.x tests on the dasboard you can see RefApp 3.x Login, RefApp 3.x Clinical Visit,etc. These names do not have the text Test on them since its already defined in the their mother card.

Its from the reason above that we as well rename tests for 2.x RefApp to such a convention to have consistency. I have already seen you embracing the ideal in how you names Search And Registarion workflow name [here ](https://github.com/openmrs/openmrs-contrib-qaframework/pull/79)

**Comment by kdaud** [comment]
> That is ticket duplication.

The reason have given [here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/90#issuecomment-905381238) is sufficient enough not  consider these two tickets duplicate.

**Comment by sherrif10** [comment]
Then we can go ahead and merge the two.

**Comment by kdaud** [comment]
> Then we can go ahead and merge the two.

No need to merge the work. Have you understood my comment [here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/90#issuecomment-905381238) ? Feel free to let me know and I explain to you if you have not gotten the idea

**Comment by sherrif10** [comment]
What could be the last opinion please, All these tickets are needed. Feel free to either close the ticket you think its not worth. Then we can work on other things please. Have left the decission in your hands.

**Comment by sherrif10** [comment]
However we would love seeing your last decission

**Comment by kdaud** [comment]
> What could be the last opinion

Have post it [here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/91#issuecomment-905457168) and would love seeing you resolving [RATEST-205](https://issues.openmrs.org/browse/RATEST-205)

**Comment by sherrif10** [comment]
> Have post it [here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/91#issuecomment-905457168) and would love seeing you resolving [RATEST-205](https://issues.openmrs.org/browse/RATEST-205)

I really consider RATEST-205   clear and precise. Feel free to do the needful please.

**Comment by kdaud** [comment]
Let me explain a few things here

Ticket ID [RATEST-202](https://issues.openmrs.org/browse/RATEST-202) 
Goal:
Refactoring Clinical Visit and Vitals-and-Triaging workflow test to have
* consistency in the workflow naming within the scripts section in [package.json|https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/package.json] file. 
* consistency in the workflow names displayed on [QA-Dashboard|https://github.com/openmrs/openmrs-contrib-qaframework]

Ticket ID [RATEST-205](https://issues.openmrs.org/browse/RATEST-205)
Goal:
To have consistency in the workflow test naming for 3.x ref app to ensure quality within our code base. 
Following the naming of clinical visit workflow name (refapp3ClinicalVisit), currently the other workflow names are labelled refApp3Login, refApp3Registration and refApp3Settings respectively.

Expected to have the workflow renamed to refapp3Login, refapp3Registration and refapp3Settings respectively

Note: There is need to make change in both in [package.json|https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/package.json] file and in their respective workflow configurations.

RATEST-202 is capturing changes for 2.x RefApp workflow tests and RATEST-205 is capturing changes for 3.x RefApp workflow tests
cc: @sherrif10

**Comment by sherrif10** [comment]
> Refactoring Clinical Visit and Vitals-and-Triaging workflow test to have

But even the way you are refactoring need improvements All the tickets are in the same module so i dont see any problem including the two in one.

**Comment by kdaud** [comment]
> But even the way you are refactoring need improvements All the tickets are in the same module so i dont see any problem including the two in one.

Yes the test workflow are present in the same module but their are affecting two different modules RefApp 2.x modules and RefApp 3.x module). My reason [here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/90#issuecomment-905381238) is sufficient to have two separate tickets.

**Comment by sherrif10** [comment]
> but their changes are affecting two different modules RefApp 2.x modules and RefApp 3.x module)

Where would these changes affect RefApp 3.x module?

**Comment by kdaud** [comment]
> Where would these changes affect RefApp 3.x module?

Its the reason why RATEST-202 is addressing changes for 2.x RefApp 

and 

RATEST-205 addressing changes for 3.x RefApp.

So we need this two tickets resolved separately based on their tickets description attached to them respectively.

**Comment by sherrif10** [comment]
> Expected to have the workflow renamed to refapp3Login, refapp3Registration and refapp3Settings respectively

I think refApp3Login, refApp3Registration and refApp3Settings seems to be more clear and readable even case sensitive should be considered. Lets hear from @jayasanka-sack @dkayiwa .

**Comment by kdaud** [comment]
> I think refApp3Login, refApp3Registration and refApp3Settings seems to be more clear and readable even case sensitive should be considered

I want to guess the comment was intended to be attached on this [PR](https://github.com/openmrs/openmrs-contrib-qaframework/pull/91). 
However, this is my take on the epic concerning with workflow names for either 2.x or 3.x.  I personally would recommend renaming them following how these modules are named on openmrs github [repos](https://github.com/openmrs). For example for 2.x its respective repository is [openmrs-distro-referenceapplication](https://github.com/openmrs/openmrs-distro-referenceapplication) so the workflow for any test would look like _refapp2TestName_ . I would wish to embrace a kind of naming the module pioneer (`author`) began with in reference to workflow test naming convention for referenceapplication as observed [here](https://github.com/openmrs/openmrs-contrib-qaframework/blob/af11d21215129d93d50fbc753dc9387881a5b74b/package.json#L11)

**Comment by sherrif10** [comment]
Its very fine @kdaud , Feel free to do the needfull please. However if its related with renaming, i would prefer case sensitive naming convention. But feel free to go ahead and commit the changes with in this ticket.

**Comment by kdaud** [comment]
> i would prefer case sensitive naming convention. But feel free to go ahead and commit the changes with in this ticket.

I personally would love a collective idea and coming up with a decision that is worth our projects that we can all consent with. Personally have given the reason why I think we should name the workflow following the existing workflow naming as stretched in my comment [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/90#issuecomment-905502918).

My major point is to have consistency within our module, if the ideal is to go with refApp2TWorkflowName, refApp3WorkflowName OR we choose to use  refapp2WorkflowName, refApp3WorkflowName then any of the two that will be selected should be embraced in all the upcoming workflows even perhaps making a refactoring in already existing workflows to match the ideal.

@dkayiwa can guide us in which workflow name could be better for our `reference application module` and whatever name that would be opted for will be glad for it as long we its embrace through out the workflows.

**Review by brandones — APPROVED** [review]

**Pull request merged**